cmd/palomad/root.go

```diff
@@ -207,3 +218,13 @@ func rootPreRunE(cmd *cobra.Command, args []string) error {
 
 	return nil
 }
+
+var tempDir = func() string {
```

Comment:
I'm aware this is some of the most horrible code you've seen in a while. It's taken straight from Cosmos SDK's reference implementation, and could likely be improved upon if I had more time. At least it works.